lib/src/Uncertainty/Algorithm/Simulation/ProbabilitySimulation.cxx

```diff
@@ -0,0 +1,123 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief ProbabilitySimulation is a generic view of simulation methods for computing
+ * probabilities and related quantities by sampling and estimation
+ *
+ *  Copyright 2005-2017 Airbus-EDF-IMACS-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/ProbabilitySimulation.hxx"
+#include "openturns/Log.hxx"
+#include "openturns/Curve.hxx"
+#include "openturns/Point.hxx"
+#include "openturns/ResourceMap.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/*
+ * @class ProbabilitySimulation
+ */
+
+CLASSNAMEINIT(ProbabilitySimulation);
+
+static const Factory<ProbabilitySimulation> Factory_ProbabilitySimulation;
+
+
+/* Constructor with parameters */
+ProbabilitySimulation::ProbabilitySimulation(const Event & event,
+                                             const WeightedExperiment & experiment,
+                                             const Bool verbose,
+                                             const HistoryStrategy & convergenceStrategy)
+  : Simulation(event, verbose, convergenceStrategy)
+{
+  setExperiment(experiment);
+}
+
+ProbabilitySimulation::ProbabilitySimulation(const Bool verbose,
+                                             const HistoryStrategy & convergenceStrategy)
+  : Simulation(verbose, convergenceStrategy)
+{
+  // Nothing to do
+}
+
+/* Virtual constructor */
+ProbabilitySimulation * ProbabilitySimulation::clone() const
+{
+  return new ProbabilitySimulation(*this);
+}
+
+
+void ProbabilitySimulation::setExperiment(const WeightedExperiment & experiment)
+{
+  experiment_ = experiment;
+  experiment_.setDistribution(getEvent().getImplementation()->getAntecedent()->getDistribution());
```

Comment:
Yes, I was writing it in parallel of the review. Here I create 4 events (one by Event constructor) and run MonteCarlo on it. With OT 1.9 I get:

> ##################################################
composite vector/comparison event
MonteCarlo result= probabilityEstimate=2.473684e-01 varianceEstimate=6.053604e-04 standard deviation=2.46e-02 coefficient of variation=9.95e-02 confidenceLength(0.95)=9.64e-02 outerSampling=95 blockSize=4
##################################################
composite vector/domain event
MonteCarlo result= probabilityEstimate=3.333333e-01 varianceEstimate=1.102293e-03 standard deviation=3.32e-02 coefficient of variation=9.96e-02 confidenceLength(0.95)=1.30e-01 outerSampling=63 blockSize=4
##################################################
composite vector/interval event
MonteCarlo result= probabilityEstimate=1.927481e-01 varianceEstimate=3.688685e-04 standard deviation=1.92e-02 coefficient of variation=9.96e-02 confidenceLength(0.95)=7.53e-02 outerSampling=131 blockSize=4
##################################################
process/domain event
MonteCarlo result= probabilityEstimate=9.166667e-01 varianceEstimate=8.101852e-03 standard deviation=9.00e-02 coefficient of variation=9.82e-02 confidenceLength(0.95)=3.53e-01 outerSampling=3 blockSize=4

and with your PR:

> ##################################################
composite vector/comparison event
WRN - class MonteCarlo is deprecated in favor of ProbabilitySimulation
MonteCarlo result= probabilityEstimate=2.473684e-01 varianceEstimate=6.053604e-04 standard deviation=2.46e-02 coefficient of variation=9.95e-02 confidenceLength(0.95)=9.64e-02 outerSampling=95 blockSize=4
##################################################
composite vector/domain event
Not supported by MonteCarlo
##################################################
composite vector/interval event
MonteCarlo result= probabilityEstimate=2.079832e-01 varianceEstimate=4.288171e-04 standard deviation=2.07e-02 coefficient of variation=9.96e-02 confidenceLength(0.95)=8.12e-02 outerSampling=119 blockSize=4
##################################################
process/domain event
Not supported by MonteCarlo

The code is:

```python
import openturns as ot
description = ot.Description()
description.add("composite vector/comparison event")
dim = 2
distribution = ot.Normal(dim)
Xvector = ot.RandomVector(distribution)
f = ot.SymbolicFunction(["x0", "x1"], ["x0+x1"])
Yvector = ot.RandomVector(f, Xvector)
s = 1.0
event1 = ot.Event(Yvector, ot.Greater(), s)
description.add("composite vector/domain event")
domain1D = ot.LevelSet(ot.SymbolicFunction(["x0"], ["sin(x0)"]), -0.5)
event2 = ot.Event(Yvector, domain1D)
vector/comparison event
description.add("composite vector/interval event")
interval = ot.Interval(0.5, 1.5)
event3 = ot.Event(Yvector, interval)
description.add("process/domain event")
Xprocess = ot.WhiteNoise(distribution, ot.RegularGrid(0.0, 0.1, 10))
domain2D = ot.LevelSet(ot.SymbolicFunction(["x0", "x1"], ["(x0-1)^2+x1^2"]), 1.0)
event4 = ot.Event(Xprocess, domain2D)
all_events = [event1, event2, event3, event4]
for i,event in enumerate(all_events):
    print "#"*50
    print description[i]
    try:
        myAlgo = ot.MonteCarlo(event)
        myAlgo.setMaximumOuterSampling(250)
        myAlgo.setBlockSize(4)
        myAlgo.setMaximumCoefficientOfVariation(0.1)
        myAlgo.run()
        print "MonteCarlo result=", myAlgo.getResult()
    except:
        print "Not supported by MonteCarlo"
```